### ISSUE TITLE
v8-compile-cache is creating a temporary directory and is not being cleaned up after running the eslint command

### ISSUE THREAD

**Issue by ylemberg** [eslint/eslint#14352]
**Tell us about your environment**

```
Environment Info:

Node version: v14.16.1
npm version: v6.14.13
Local ESLint version: Not found
Global ESLint version: v7.24.0 (Currently used)
Operating System: win32 10.0.19042
```

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
None

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
C:\> cd .\temp\
C:\temp> npm install -g eslint
C:\temp> ls #empty directory displays
C:\temp> eslint
C:\temp> ls #new directory "fadsfs" appears
C:\temp> ls .\fadsfs\v8-compile-cache\8.4.371.19-node.18\ #Displays BLOB and MAP files exists
```
Screenshot of reproduced flow
![2021-04-22 17_16_47-C__temp ~ PowerShell 5 1 19041 906 64-bit (2860)](https://user-images.githubusercontent.com/5579150/115800459-84499e00-a38f-11eb-9942-55552eca6adc.png)

**What did you expect to happen?**
Any temporary directories created to be cleaned out once the `eslint` command completes

**What actually happened? Please include the actual, raw output from ESLint.**
A temporary directory created by v8-compile-cache remains after the eslint lint command completes (please see above for raw output from eslint)

- If I comment out [this line of code](https://github.com/eslint/eslint/blob/master/bin/eslint.js#L13), the issue does not reproduce. Confirming my suspicions that when eslint runs v8-compile-cache, it is creating this directory.
- Running v8-compile-cache standalone does not reproduce the issue. If I make an index.js file that only has `require('v8-compile-cache')` for it's contents and run `node index.js` , the issue does not reproduce
- I've uninstalled nodejs completely [following these steps](https://stackoverflow.com/a/20711410) and then reinstalled nodejs and the issue persists
- I've tried using several different versions of npm and nodejs and the issue persists

Any insight as to what in my environment might be causing eslint to create these folders would be greatly apprecaited. Thank you.

**Comment by nzakas** [comment]
This behavior is how v8-compile-cache works. It’s intended to speed up execution time. See: https://github.com/zertosh/v8-compile-cache

is there a reason you’re concerned about this temp directory?

**Comment by ylemberg** [comment]
I am not concerned about a temporary directory being created. But this temporary directory is persisting and not being cleaned up/removed.

**Comment by aladdin-add** [comment]
To disable the cache, please set the environment variable DISABLE_V8_COMPILE_CACHE=1.
https://github.com/zertosh/v8-compile-cache#options

**Comment by ylemberg** [comment]
I've attempted this solution. However, I am also using [react-scripts](https://github.com/facebook/create-react-app/blob/master/packages/cra-template/template/README.md#npm-start) in my project and when I run their commands it is now throwing an error looking for the directory that it thinks v8-compile-cache created. Below is a screenshot of `react-scripts start` erroring because the `fadsfs` directory does not exist that it thinks v8-compile-cache created when running eslint
![2021-04-23 11_39_19-cosmos  master  ~ PowerShell 5 1 19041 906 64-bit (20644)](https://user-images.githubusercontent.com/5579150/115915748-99264000-a428-11eb-8c09-c5915523428f.png)

**Comment by ylemberg** [comment]
Oh wow, ok so I figured out the issue. In case anyone else has a similar problem I'll document what fixed it for me here.

I guess what happened is the "TEMP" environment variable got removed somehow (I almost never edit ENV Vars so not sure how this happened). And then when I was running eslint -> which runs v8-compile-cache -> it looked for the env var "TEMP", but nothing was there so it defaulted to making a new temp directory in my current directory. So I incorrectly assumed this was supposed to be cleaned out, but in reality the directory was just being pointed to my current directory instead of the accurate directory of C:/Users/<User>/AppData/Local/Temp.

Closing the issue now.
![2021-04-23 11_33_14-Environment Variables](https://user-images.githubusercontent.com/5579150/115917876-a264dc00-a42b-11eb-9585-0927fc0559a9.png)
